ray-operator/controllers/ray/common/route.go

```diff
@@ -0,0 +1,83 @@
+package common
+
+import (
+	"github.com/ray-project/kuberay/ray-operator/controllers/ray/utils"
+
+	routev1 "github.com/openshift/api/route/v1"
+	rayv1alpha1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// BuildRouteForHeadService Builds the Route (OpenShift) for head service dashboard.
+// This is used to expose dashboard and ray job submission API for external traffic.
+func BuildRouteForHeadService(cluster rayv1alpha1.RayCluster) (*routev1.Route, error) {
+	labels := map[string]string{
+		RayClusterLabelKey:                cluster.Name,
+		RayIDLabelKey:                     utils.GenerateIdentifier(cluster.Name, rayv1alpha1.HeadNode),
+		KubernetesApplicationNameLabelKey: ApplicationName,
+		KubernetesCreatedByLabelKey:       ComponentName,
+	}
+
+	// Copy other route configurations from cluster annotations to provide a generic way
+	// for user to customize their route settings. 
+	// The `exclude_set` is used to avoid setting
+	// both IngressClassAnnotationKey annotation which is deprecated and `Spec.IngressClassName`
+	// at the same time.
+	exclude_set := map[string]struct{}{
+		IngressClassAnnotationKey: {},
```

Comment:
This annotation appears to be specific to "ingress". Please remove it if "route" does not have the convention of the `kubernetes.io/ingress.class` annotation.

History context:

1. Very old Kubernetes: Use the `kubernetes.io/ingress.class` annotation to specify the ingress controller being used, such as nginx.

2. Old Kubernetes: Both `kubernetes.io/ingress.class` annotation and `Spec.IngressClassName` can be used to specify the ingress controller being used.

3. Now: The `kubernetes.io/ingress.class` annotation is deprecated, and an error will be reported if both the annotation and `Spec.IngressClassName` are specified.

If OpenShift route does not have these conventions, please remove `IngressClassAnnotationKey` (i.e. `kubernetes.io/ingress.class`). Thanks!

Reply:
Done